### PR TITLE
Team ceres/feature/terra form custom container

### PR DESCRIPTION
**Pull request by felixdausch**
@plentymarkets/team-terra

### Definition of Done
Documentation
- [x] Changelog
- [x] Example (updated or created)
- [x] JSDoc

Testing
- [x] Unit Test successful
- [x] Person 1
- [x] Person 2

Information transfer
- [x] Inform Ceres about changes in `TerraFormComponent`

### CODE REVIEW

File: src/lib/components/forms/form/form-container/container-wrapper/terra-form-container-wrapper.component.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+ * Wraps custom components around terra-form-container component.
+ * This enables usage of structural components to render nested forms
+ */
+@Component({
+    selector: 'terra-form-container-wrapper',
+    template: require('./terra-form-container-wrapper.component.html')
+})
+export class TerraFormContainerWrapperComponent extends TerraFormEntryBase implements OnInit, OnChanges, OnDestroy
+{
+    @Input()
```
Comment (dantrau) [inline src/lib/components/forms/form/form-container/container-wrapper/terra-form-container-wrapper.component.ts]: please add description to inputs

File: src/lib/components/forms/form/form-container/container-wrapper/terra-form-container-wrapper.component.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    public inputScope:TerraFormScope;
```
Comment (dantrau) [inline src/lib/components/forms/form/form-container/container-wrapper/terra-form-container-wrapper.component.ts]: Inputs are not prefixed with ìnput` anymore

File: src/lib/components/forms/form/form-container/container-wrapper/terra-form-container-wrapper.component.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    @Input()
+    public inputFormGroup:FormGroup;
```
Comment (dantrau) [inline src/lib/components/forms/form/form-container/container-wrapper/terra-form-container-wrapper.component.ts]: remove prefix

File: src/lib/components/forms/form/form-entry/terra-form-entry.base.ts [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+} from '../../../..';
+import { FormEntryContainerDirective } from './form-entry-container.directive';
+
+export class TerraFormEntryBase implements OnChanges, OnDestroy
+{
+    /**
+     * @description Specification of the formField that should be displayed.
+     */
+    @Input()
+    public inputFormField:TerraFormFieldInterface;
```
Comment (dantrau) [inline src/lib/components/forms/form/form-entry/terra-form-entry.base.ts]: remove prefix

File: src/lib/components/forms/form/form-entry/terra-form-entry.base.ts [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+    /**
+     * @description Map of supported control types. If the given formField's type is not supported, a TerraTextInputComponent instance is
+     *     rendered by default.
+     *     Please note: All of the control types contained in this map have to implement the ControlValueAccessor interface.
+     * @default {} - an empty map. Hence, not a single control type is supported and the default type TerraTextInputComponent will be
+     *     rendered as well.
+     */
+    @Input()
+    public inputControlTypeMap:{ [key:string]:Type<any> | TerraFormTypeInterface } = {};
```
Comment (dantrau) [inline src/lib/components/forms/form/form-entry/terra-form-entry.base.ts]: remove prefix

File: src/lib/components/forms/form/form-entry/terra-form-entry.base.ts [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+
+    /**
+     * @description May be used to disable/enable the form field.
+     * @default false
+     */
+    @Input()
+    public inputIsDisabled:boolean = false;
```
Comment (dantrau) [inline src/lib/components/forms/form/form-entry/terra-form-entry.base.ts]: remove prefix

File: src/lib/components/forms/form/form-container/container-wrapper/terra-form-container-wrapper.component.ts (hunk 2 above)
Comment (dantrau) [reply src/lib/components/forms/form/form-container/container-wrapper/terra-form-container-wrapper.component.ts]: resolved: avoidance of breaking changes

File: src/lib/components/forms/form/form-entry/terra-form-entry.base.ts (hunk 4 above)
Comment (dantrau) [reply src/lib/components/forms/form/form-entry/terra-form-entry.base.ts]: resolved: avoidance of breaking changes

File: src/lib/components/forms/form/form-container/container-wrapper/terra-form-container-wrapper.component.ts (hunk 3 above)
Comment (dantrau) [reply src/lib/components/forms/form/form-container/container-wrapper/terra-form-container-wrapper.component.ts]: resolved: avoidance of breaking changes

File: src/lib/components/forms/form/form-entry/terra-form-entry.base.ts (hunk 6 above)
Comment (dantrau) [reply src/lib/components/forms/form/form-entry/terra-form-entry.base.ts]: resolved: avoidance of breaking changes

File: src/lib/components/forms/form/form-container/container-wrapper/terra-form-container-wrapper.component.ts (hunk 1 above)
Comment (felixdausch) [reply src/lib/components/forms/form/form-container/container-wrapper/terra-form-container-wrapper.component.ts]: ✅

File: src/lib/components/forms/form/form-container/container-wrapper/terra-form-container-wrapper.component.html [hunk 7]
```diff
@@ -0,0 +1 @@
+<div tcFormEntryContainer></div>
```
Comment (MFrank84) [inline src/lib/components/forms/form/form-container/container-wrapper/terra-form-container-wrapper.component.html]:
```suggestion
<div tcFormEntryContainer></div>

```

File: src/lib/components/forms/form/form-container/container-wrapper/terra-form-container-wrapper.component.ts [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+    OnChanges,
+    OnDestroy,
+    OnInit,
+    SimpleChanges
+} from '@angular/core';
+import {
+    TerraFormContainerComponent,
+    TerraFormScope,
+    TerraPortletComponent
+} from '../../../../..';
```
Comment (MFrank84) [inline src/lib/components/forms/form/form-container/container-wrapper/terra-form-container-wrapper.component.ts]: Resolve barrel imports

File: src/lib/components/forms/form/form-entry/terra-form-entry.base.ts [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+    OnDestroy,
+    SimpleChanges,
+    Type,
+    ViewChild
+} from '@angular/core';
+import { TerraFormTypeInterface } from '../model/terra-form-type.interface';
+import {
+    TerraFormFieldInterface,
+    TerraTextInputComponent
+} from '../../../..';
```
Comment (MFrank84) [inline src/lib/components/forms/form/form-entry/terra-form-entry.base.ts]: Resolve barrel imports

File: src/lib/components/forms/form/model/form-type-map.ts [hunk 10]
```diff
@@ -14,6 +14,7 @@ import { TerraColorPickerComponent } from '../../input/color-picker/terra-color-
 import { TerraNoteEditorComponent } from '../../../editors/note-editor/terra-note-editor.component';
 import { TerraCodeEditorComponent } from '../../../editors/code-editor/terra-code-editor.component';
 import { TerraSliderComponent } from '../../slider/terra-slider.component';
+import { TerraPortletComponent } from '../../../..';
```
Comment (MFrank84) [inline src/lib/components/forms/form/model/form-type-map.ts]: Resolve barrel import

**Pull request merged**